src/Npgsql/NpgsqlTypes/NpgsqlDate.cs

```diff
@@ -1,452 +1,235 @@
-using System;
-using System.Collections;
-using System.Collections.Generic;
-using System.Text;
+﻿using System;
 
-#pragma warning disable 1591
-
-// ReSharper disable once CheckNamespace
 namespace NpgsqlTypes
 {
-    [Serializable]
-    public readonly struct NpgsqlDate : IEquatable<NpgsqlDate>, IComparable<NpgsqlDate>, IComparable,
-        IComparer<NpgsqlDate>, IComparer
+    /// <summary>
+    /// Represents a date.
+    /// </summary>
+    public readonly struct NpgsqlDate : IEquatable<NpgsqlDate>, IComparable<NpgsqlDate>
     {
-        //Number of days since January 1st CE (January 1st EV). 1 Jan 1 CE = 0, 2 Jan 1 CE = 1, 31 Dec 1 BCE = -1, etc.
-        readonly int _daysSinceEra;
-        readonly InternalType _type;
-
-        #region Constants
-
-        static readonly int[] CommonYearDays = { 0, 31, 59, 90, 120, 151, 181, 212, 243, 273, 304, 334, 365 };
-        static readonly int[] LeapYearDays = { 0, 31, 60, 91, 121, 152, 182, 213, 244, 274, 305, 335, 366 };
-        static readonly int[] CommonYearMaxes = { 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 };
-        static readonly int[] LeapYearMaxes = { 31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 };
+        readonly int _days;
 
         /// <summary>
-        /// Represents the date 1970-01-01
+        /// Initializes a new instance of the <see cref="NpgsqlDate"/>
+        /// structure to a specified number of days since PostgreSQL epoch.
         /// </summary>
-        public static readonly NpgsqlDate Epoch = new(1970, 1, 1);
+        /// <param name="days">
+        /// A date expressed in number of days since PostgreSQL epoch.
+        /// </param>
+        /// <seealso cref="Days"/>
+        /// <seealso cref="PostgreSqlEpoch"/>
+        public NpgsqlDate(int days) => _days =
```

Comment:
As per my general comment, I think this should be removed.

Reply:
Removed means no constructors to make that type at all from raw data, so perhaps, you wanted to say "hidden", right? Am fine with it since don't like to expose internals like BCL date time types do. If it would be required we can provide `TryRead`/`TryWrite` methods accepting a span.

Reply:
Yeah, I'm only referring to this being public. Of course it's fine for it to be internal so that a handler can construct it.